dds/DCPS/RTPS/BaseMessageUtils.h

```diff
@@ -139,15 +142,23 @@ locator_to_address(ACE_INET_Addr& dest, const OpenDDS::DCPS::Locator_t& locator)
     break;
 #endif
   case LOCATOR_KIND_UDPv4:
+#if !defined (ACE_HAS_IPV6) && !defined (IPV6_V6ONLY)
```

Comment:
On this line, replace && with || (inverse logic of the #if below).
Other lines with ACE_UNUSED_ARG should not be needed.